src/nodes/hypertable_modify.c

```diff
@@ -752,6 +752,11 @@ ExecModifyTable(CustomScanState *cs_node, PlanState *pstate)
 	 */
 	if ((operation == CMD_DELETE || operation == CMD_UPDATE) && !ht_state->comp_chunks_processed)
 	{
+		if (!ts_guc_enable_dml_decompression)
```

Comment:
I think when ts_guc_enable_dml_decompression is set to false, even UPDATE/DELETE on regular hyperatable will fail. We don't want this behaviour.
IMO we should report error only when the chunk which is getting affected is compressed.

Best place to report error is in decompress_target_segments() before we call decompress_batches_for_update_delete().

Reply:
I agree it seems like we have to push it down a bit until we can assert we are dealing with compressed chunks.

Reply:
You're correct. Fixed it and also added more tests creating new uncompressed chunk and executing UPDATE/DELETE on it... also testes UPDATE/DELETE with conditions that cross compressed and uncompressed chunks.